library/Makefile

```diff
@@ -39,6 +39,8 @@ LOCAL_CFLAGS += -fPIC -fpic
 endif
 endif
 
+LOCAL_CFLAGS += -DALLOW_PRIVATE_ACCESS
```

Comment:
Having to define an additional macro to compile the library is very unfriendly. Instead, library source files should include a header that defines `ALLOW_PRIVATE_ACCESS` before it includes any other header. We already have such a header: `common.h`.

Reply:
Removed ALLOW_PRIVATE_ACCESS from build files, renamed it to MBEDTLS_ALLOW_PRIVATE_ACCESS and placed it's definition inside common.h